server.js

```diff
@@ -91,13 +102,26 @@ const openApiContents = yaml.load(fs.readFileSync("openapi.yaml"));
 // Matomo
 const apiTracker = lib.setupTracker(openApiContents);
 
-// Detailed processes files
+// By default get the detailed path from the env, otherwise consider that the detailed
+// files are stored in the current public directory
+let detailedProcessesDirectory = ECOBALYSE_DATA_DIR || "./public";
+
+let textileImpactsFile = `${detailedProcessesDirectory}/data/textile/processes_impacts.json`;
+let foodImpactsFile = `${detailedProcessesDirectory}/data/food/processes_impacts.json`;
 
-const textileImpactsFile = "public/data/textile/processes_impacts.json";
-const foodImpactsFile = "public/data/food/processes_impacts.json";
 const textileFile = "public/data/textile/processes.json";
 const foodFile = "public/data/food/processes.json";
 
+// If detailed impacts files doesn't exist,
+// we should default to the non detailed files
+if (!fs.existsSync(textileImpactsFile)) {
+  textileImpactsFile = textileFile;
+}
+
+if (!fs.existsSync(foodImpactsFile)) {
+  foodImpactsFile = foodFile;
+}
+
```

Comment:
Note that this wouldn't be required anymore with the new suggested `checkDataFile` implementation.